sql/parser/eval.go

```diff
@@ -259,6 +263,147 @@ func (d DString) String() string {
 	return StrVal(d).String()
 }
 
+// DDate is the date Datum.
+type DDate struct {
```

Comment:
Right, I agree. I'd rather use a real type alias and be explicit about
passthrough methods.
On Sep 1, 2015 12:20, "Tobias Schottdorf" <notifications@github.com> wrote:

> In sql/parser/eval.go
> <https://github.com/cockroachdb/cockroach/pull/2314#discussion_r38438844>:
>
> > @@ -259,6 +263,147 @@ func (d DString) String() string {
> >  	return StrVal(d).String()
> >  }
> >
> > +// DDate is the date Datum.
> > +type DDate struct {
>
> I'm sure that's intentional, though I don't know what the "killer
> argument" is. Being able to re-define existing methods on the type "alias"
> (Pete doesn't like to call it that, and I tend to think he's right) is one
> advantage (among other things, this allows exposing only a subset of the
> original methods by simply passing through).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/2314/files#r38438844>.
>